src/Engine/RadiumEngine.hpp

```diff
@@ -214,18 +314,40 @@ class RA_ENGINE_API RadiumEngine
     std::string m_resourcesRootDir;
 
     /// active fbo and viewport management, this class handle fbo and viewport
-    /// values so that one can save and restore these value with pushFboAndViewport and popFboAndViewport
+    /// values so that one can save and restore these value with pushFboAndViewport and
+    /// popFboAndViewport
     class FboAndViewport
     {
       public:
-        FboAndViewport( int fbo, std::array<int,4>&& viewport ) :
-            m_fbo{fbo}, m_viewport{viewport} {}
+        FboAndViewport( int fbo, std::array<int, 4>&& viewport ) :
+            m_fbo{fbo},
+            m_viewport{viewport} {}
         int m_fbo;
-        std::array<int,4> m_viewport;
+        std::array<int, 4> m_viewport;
     };
 
     /// Stack of saved fbo and viewport values @see pushFboAndViewport popFboAndViewport
     std::stack<FboAndViewport> m_fboAndViewportStack;
+
+    struct TimeData {
+        /**
+         * Update the current time from the current frame number, according to the
+         * time flow policy (ForwardBackward or loop around).
+         */
+        void updateTime();
```

Comment:
As said in the comment (and in the code of the method), if this method ensures some constraints (or invariant) between members, why are they public ? 
public members can be changed without ensuring any constraint (or invariant) between them.
If there is some constraints between members (and there are), put the constrained members (e.g. `m_time`) in a private section.
Even if this class is private to the Engine, it is important to ensure correct usage and data coherency.

Reply:
There is no specific constraint, the time window can be whatever, same for the modes.
Only *m_time*, which can still be whatever when time flow is on hold, has to be remapped into the time window when time flows. I don't see the point of having it private with setter and getter.

Reply:
I see one reason (from the code of the function update time) : `m_time` is always in the interval `[start, end]`. This is an invariant that must be ensured when setting time (or start, or end) and not at each update.

Reply:
This invariant has to be met only when time flows.
Let's say I have an animation from t=0 to t=10 (in seconds), but I just want to play it from t=2 to t=8 and edit that part.
I should still be able to see what happens outside [2,8] for convenience checks or whatever, for example using the keyframe from t=0 for t=7.
That's why I can set the time to any value, as soon as time flow is stopped.